gradle/baseline.gradle

```diff
@@ -17,12 +17,18 @@ List<String> blacklistedBaselineProjects = [
         'profile-client-protobufs',
         'timestamp-impl']
 
-
 if (!blacklistedBaselineProjects.contains(project.name)) {
     apply plugin: 'com.palantir.baseline-checkstyle'
 
-    // We are not presently using errorprone because that swaps out the compiler to one based on the beta compiler
-    // for JDK 9, which has a known issue with varargs.
+    // ./gradlew compileJava compileTestJava -PenableErrorProne=true
+    if (enableErrorProne.toBoolean()) {
+        apply plugin: 'org.inferred.processors'  // installs the "processor" configuration needed for baseline-error-prone
+        apply plugin: 'com.palantir.baseline-error-prone'
+
+        configurations.errorprone {
+            resolutionStrategy { force('com.google.guava:guava:21.0') }
```

Comment:
I'm not really sure. It's not documented by the error-prone docs, but seems to be a widespread practice regarding our internal usage. It was added on #1853, I'm just adding it back. Maybe @gsheasby has a better understanding of this.